### PR TITLE
One way coupling MF and heat transfer

### PR DESCRIPTION
**Pull request by lpsaavedra**
# Description of the problem

The `lethe-fluid-matrix-free` application was not coupled to any other physics. 

# Description of the solution

This PR enables the one way coupling between fluid dynamics and heat transfer using the existent multiphysics interface.

# How Has This Been Tested?
One test was added to verify the coupling. It is the same one that is used in the matrix-based application:
- [X] `applications_tests/lethe-fluid-matrix-free/heat_transfer_high_pe_stab_gls.prm`

### CODE REVIEW

File: include/solvers/mf_navier_stokes.h [hunk 1]
```diff
@@ -420,6 +419,13 @@ class MFNavierStokesSolver
   void
   print_mg_setup_times();
 
+
```
Comment (voferreira) [inline include/solvers/mf_navier_stokes.h:422]:
```suggestion
```

**Review by voferreira — APPROVED** [review]
All clear. I don't think I have anything to add. Thank you, Laura and Mikael, for the great work.

File: applications_tests/lethe-fluid-matrix-free/heat_transfer_high_pe_stab_gls.prm [hunk 2]
```diff
@@ -0,0 +1,194 @@
+# Listing of Parameters
+#----------------------
+
+set dimension = 2
+
+#---------------------------------------------------
```
Comment (blaisb) [inline applications_tests/lethe-fluid-matrix-free/heat_transfer_high_pe_stab_gls.prm:6]: I think it was a good idea to make a unit test with this. Great thinking.

File: source/solvers/mf_navier_stokes.cc [hunk 3]
```diff
@@ -44,6 +44,24 @@
...
 #include <deal.II/numerics/vector_tools.h>
 
+/**
+ * @brief Helper function that allows to convert deal.II vectors to Trilinos vectors.
+ *
+ * @tparam number Abstract type for number across the class (i.e., double).
+ * @param out Destination TrilinosWrappers::MPI::Vector vector.
+ * @param in Source LinearAlgebra::distributed::Vector<number> vector.
+ */
+template <typename number>
```
Comment (blaisb) [inline source/solvers/mf_navier_stokes.cc:54]: Could this be moved outside of the MF files? I feel like this could be useful in many placed

**Review by blaisb — APPROVED** [review]
1-2 minor comments to address. Great work !!!

File: source/solvers/mf_navier_stokes.cc (hunk 3 above)
Comment (lpsaavedra) [reply source/solvers/mf_navier_stokes.cc:54]: We could move it but I think we should leave it here for now... this is the only place where it is used and eventually we will remove it once multiphysics support also `LinearAlgebra::distributed::Vector`

File: include/solvers/mf_navier_stokes.h [hunk 4]
```diff
@@ -420,6 +419,12 @@ class MFNavierStokesSolver
   void
   print_mg_setup_times();
 
+  /**
+   * @brief  Provide relevant solutions to multiphysics interface.
```
Comment (mivaia) [inline include/solvers/mf_navier_stokes.h:423]:
```suggestion
   * @brief  Provide present and previous flow solutions to the multiphysics interface. These solutions can be accessed through the multiphysics interface by the different physics. 
```

**Review by mivaia — APPROVED** [review]
Nice job! Only a single comment :)

File: source/solvers/mf_navier_stokes.cc (hunk 3 above)
Comment (blaisb) [reply source/solvers/mf_navier_stokes.cc:54]: OKidoke

**Pull request merged**